Tests/ReactiveSwiftTests/SignalProducerSpec.swift

```diff
@@ -1970,6 +1988,7 @@ class SignalProducerSpec: QuickSpec {
 				producer
 					.observe(on: TestScheduler())
 					.startWithSignal { signal, innerDisposable in
+						signal.observe { _ in }
```

Comment:
Under the new signal lifetime semantic, a signal is kept alive only if it is retained or has active observers. So the observer is added to keep the signal alive, or otherwise the signal would dispose of itself upon exiting the scope.

The bug fixed is kinda related too - a produced signal would dispose of itself when it is not retained and not observed. But then its associated producer disposable would not follow.